packages/govuk-frontend/src/govuk/common/configuration.mjs

```diff
@@ -0,0 +1,334 @@
+import { ConfigError } from '../errors/index.mjs'
+import { GOVUKFrontendComponent } from '../govuk-frontend-component.mjs'
+
+import { isObject, formatErrorMessage } from './index.mjs'
+
+/**
+ * Base Component class
+ *
+ * Centralises the behaviours shared by our components
+ *
+ * @virtual
+ * @template {ObjectNested} [ConfigurationType={}]
+ * @template {Element} [RootElementType=HTMLElement]
```

Comment:
**question** If we were to update that type to `{Element & { dataset: DOMStringMap }}` would it remove the need to cast [on line 55](https://github.com/alphagov/govuk-frontend/pull/5475/files#diff-b2396d30ecca431162c287335bd70ad2314da293a2e8cfe33271a66828efee07R55)?

Reply:
Yes, we can! Nice spot.